packages/mantine/src/components/button/Button.tsx

```diff
@@ -1,27 +1,21 @@
-import {Button as MantineButton, ButtonProps as MantineButtonProps, useMantineTheme} from '@mantine/core';
+import {Button as MantineButton, ButtonProps as MantineButtonProps} from '@mantine/core';
 import {forwardRef} from 'react';
 
 import {createPolymorphicComponent} from '../../utils';
 import {ButtonWithDisabledTooltip, ButtonWithDisabledTooltipProps} from './ButtonWithDisabledTooltip';
 
 export interface ButtonProps extends MantineButtonProps, ButtonWithDisabledTooltipProps {}
 
-const _Button = forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
-    const theme = useMantineTheme();
-    return (
+const _Button = forwardRef<HTMLButtonElement, ButtonProps>(
+    ({disabledTooltip, disabled, disabledTooltipProps, ...others}, ref) => (
         <ButtonWithDisabledTooltip
-            component={MantineButton}
-            ref={ref}
-            disabledHoverColor={theme.colors.gray[2]}
-            {...props}
-        />
-    );
-});
+            disabled={disabled}
+            disabledTooltip={disabledTooltip}
+            disabledTooltipProps={disabledTooltipProps}
+        >
+            <MantineButton ref={ref} disabled={disabled} {...others} />
+        </ButtonWithDisabledTooltip>
+    )
+);
 
-export const Button = createPolymorphicComponent<
-    'button',
-    ButtonProps,
-    {Group: typeof MantineButton.Group; DisabledTooltip: typeof ButtonWithDisabledTooltip}
```

Comment:
I figured the button with disabled tooltip component shouldn't be exported. We should keep it for internal use only since its hard to understand what it actually is for.